### ISSUE TITLE
Bun Dependency Order for running scripts via --filter does not work

### ISSUE THREAD

**Issue by m11hauk** [oven-sh/bun#13239]
### What version of Bun is running?

1.1.21

### What platform is your computer?

Linux 5.15.133.1-microsoft-standard-WSL2 x86_64 unknown

### What steps can reproduce the bug?

My specific scenario is a simple one package builds a 'dist' whereas the other package depends on that being already build. But the below example should demonstrate sufficiently.

To reproduce:
1. clone this example workspace repo https://github.com/m11hauk/bun-issue-13239
2. run `bun clean` & `bun i` just to be safe
3. run `bun --filter '*' build`

### What is the expected behavior?

As said in the doc:

> Bun will respect package dependency order when running scripts. Say you have a package foo that depends on another package bar in your workspace, and both packages have a build script. When you run bun --filter '*' build, you will notice that foo will only start running once bar is done.

### What do you see instead?

This does not appear to work. Both scripts are executed at the same time, and the 'build' of the package-b finished earlier than it's dependency, tha package-a.

Edit:

Added a custom repo to demonstrate the issue.

**Comment by Hamza12700** [comment]
In bun `v1.1.22` works just fine. Update to version: 1.1.22

**Comment by chimericdream** [comment]
Do the scripts have `pre-` and/or `post-` hooks? If so, it's possible this is the same issue I reported in April (#10357). In my case, it looks like Bun is losing track of which hooks have finished and whether a given script is safe to run.

**Comment by m11hauk** [comment]
@chimericdream no they do not, I don't think the issue is in pre/post hooks. However I think that the issue you reported is dealing with the same problem, which means that the order at which the scripts are run in a situation where:
* monorepo is present
* packages depend on each other
* scripts with the same name are run from the root via the --filter option

The scripts are run simultaneously instead of being run in order of the inter package dependencies as mentioned in the doc. I think that might solve your underlying issue.

I created a simple repo for demonstration
https://github.com/m11hauk/bun-issue-13239

**Comment by Hamza12700** [comment]
@m11hauk I wasn't on the same page as you. I tested it again using bun version `1.1.25 canary build` and yes scripts `a` and `b` starts simultaneously, without any dependency order.

**Comment by kevinmarrec** [comment]
I still have the issue with `1.2.0`, and it seems this issue is a duplicate of https://github.com/oven-sh/bun/issues/12203.

https://bun.sh/docs/cli/filter#dependency-order is just not working
